### PR TITLE
update(Price, PriceComparison, PriceGroup): Handle invalid currency values.

### PR DESCRIPTION
**Pull request by stefhatcher**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

if and invalid currency like`"[Hidden]"` is given, render `Empty`.

## Motivation and Context

more bad values. follow up of https://github.com/airbnb/lunar/pull/366

## Testing

- storybook
- added tests

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/core/src/components/Price/index.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ function Price({
   precision = 0,
   round,
   trimTrailingZeros,
-}: PriceProps) {
+}: PriceProps): JSX.Element | null {
```
Comment (milesj) [inline packages/core/src/components/Price/index.tsx:41]: This doesnt seem like it should be necessary?

Comment (stefhatcher) [reply packages/core/src/components/Price/index.tsx:41]: ah, might've been while i was trying something. will double check

File: packages/core/src/utils/formatPrice.ts [hunk 2]
```diff
@@ -31,30 +31,38 @@ export default function formatPrice(
...
+      amountStr = `(${amountStr})`;
+    }
+
+    return amountStr;
+  } catch (error) {
+    // if (__DEV__) {
+    //   console.error(error);
+    // }
+
+    return '';
```
Comment (milesj) [inline packages/core/src/utils/formatPrice.ts:66]: How much would break if we updated this to be null?

Comment (stefhatcher) [reply packages/core/src/utils/formatPrice.ts:66]: @milesj a good chunk. ran into issues with `null` when doing the datetime stuff. went with `undefined` and then had to do a lot of type updates in our apps to handle that :/

Comment (milesj) [reply packages/core/src/utils/formatPrice.ts:66]: I'm just worried that downstream consumers wouldn't be aware that invalid = empty string.

Comment (stefhatcher) [reply packages/core/src/utils/formatPrice.ts:66]: good point! yeah, let's change it to `null` then

**Pull request merged**